### PR TITLE
API diff between .NET 8 Preview 6 and .NET 8 Preview 7

### PR DESCRIPTION
**Pull request by carlossanlop**
**Known issues**:
- The AsmDiff tool sometimes skips adding `{` or `{}`. Please provide a GitHub suggestion in this PR so I can correct it. [arcade#10981](https://github.com/dotnet/arcade/issues/10981)
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect. I have an issue to track this: [arcade#13814](https://github.com/dotnet/arcade/issues/13814) 
  
Repos area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @dreddy-work @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @jeffhandley @karelz @ericstj @stephentoub

Libraries area owners:
 
- System @dotnet/area-system-runtime 
- System.Collections @dotnet/area-system-collections 
- System.Diagnostics.Metrics @tommcdon @noahfalk 
- System.Net @dotnet/ncl 
- System.Numerics @dotnet/area-system-numerics 
- System.Reflection @dotnet/area-system-reflection 
- System.Runtime.CompilerServices: @dotnet/area-system-runtime-compilerservices 
- System.Runtime.InteropServices @dotnet/interop-contrib
- System.Runtime.Intrinsics @dotnet/area-system-runtime-intrinsics @lewing @radekdoulik 
- System.Security @dotnet/area-system-security 
- System.Text.Json @dotnet/area-system-text-json 
- System.Text @dotnet/area-system-text-encoding 
- System.Xml @dotnet/area-system-xml 

- Microsoft.Extensions.DependencyInjection @dotnet/area-extensions-dependencyinjection 
- Microsoft.Extensions.Diagnostics.Metrics @tommcdon @noahfalk 
- Microsoft.Extensions.Hosting @dotnet/area-extensions-hosting 
- Microsoft.Extensions.Http.Logging @dotnet/area-extensions-logging @dotnet/ncl
- Microsoft.Extensions.Options @dotnet/area-extensions-options

### CODE REVIEW

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Numerics.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# System.Numerics
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Numerics.md:1]: System.Numerics LGTM.

It's namely the addition of the `IUtf8SpanParsable` interface

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Runtime.CompilerServices.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# System.Runtime.CompilerServices
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Runtime.CompilerServices.md:1]: System.Runtime.CompilerServices LGTM. New attributes were exposed so Roslyn doesn't have to generate internal polyfills in every assembly.

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Runtime.Intrinsics.Wasm.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# System.Runtime.Intrinsics.Wasm
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Runtime.Intrinsics.Wasm.md:1]: System.Runtime.Intrinsics.Wasm LGTM

CC. @radekdoulik

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
+# System
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.md:1]: System LGTM.

Namely addition of the `IUtf8SpanParsable` interface

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Collections.Frozen.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+-        public static FrozenDictionary<TKey, TValue> ToFrozenDictionary<TKey, TValue>(this IEnumerable<KeyValuePair<TKey, TValue>> source, IEqualityComparer<TKey>? comparer, bool optimizeForReading);
+
+     }
+     public static class FrozenSet {
+-        public static FrozenSet<T> ToFrozenSet<T>(this IEnumerable<T> source, bool optimizeForReading);
+
+-        public static FrozenSet<T> ToFrozenSet<T>(this IEnumerable<T> source, IEqualityComparer<T>? comparer, bool optimizeForReading);
+
+     }
```
Comment (stephentoub) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Collections.Frozen.md:16]: LGTM

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Collections.Immutable.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
++        public static ImmutableQueue<T> Create<T>(ReadOnlySpan<T> items);
+     }
+     public static class ImmutableSortedSet {
++        public static ImmutableSortedSet<T> Create<T>(IComparer<T>? comparer, ReadOnlySpan<T> items);
++        public static ImmutableSortedSet<T> Create<T>(ReadOnlySpan<T> items);
+     }
+     public static class ImmutableStack {
++        public static ImmutableStack<T> Create<T>(ReadOnlySpan<T> items);
+     }
+ }
```
Comment (stephentoub) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Collections.Immutable.md:22]: LGTM

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Text.Unicode.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# System.Text.Unicode
+
+``` diff
+ namespace System.Text.Unicode {
+     public static class Utf8 {
++        public static bool IsValid(ReadOnlySpan<byte> value);
```
Comment (stephentoub) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Text.Unicode.md:6]: LGTM

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Diagnostics.Metrics.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+``` diff
+ namespace System.Diagnostics.Metrics {
++    public interface IMeterFactory : IDisposable {
++        Meter Create(MeterOptions options);
++    }
++    public static class MeterFactoryExtensions {
++        public static Meter Create(this IMeterFactory meterFactory, string name, string? version = null, IEnumerable<KeyValuePair<string, object?>>? tags = null);
++    }
+ }
```
Comment (tarekgh) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Diagnostics.Metrics.md:11]: I think these moved interfaces from one assembly to other assembly. but existed in preview 5 as I recall.

**Comment by tarekgh** [comment]
`System.Diagnostics` LGTM. I put a comment on `System.Diagnostics.Metrics` just in case need to address?

**Review by tannergooding — APPROVED** [review]

**Review by eiriktsarpalis — APPROVED** [review]

**Review by adamsitnik — APPROVED** [review]
FrozenCollections ctor changes LGTM

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Diagnostics.Metrics.md (hunk 8 above)
Comment (carlossanlop) [reply release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Diagnostics.Metrics.md:11]: Was that movement expected, or are you saying the API diff tool has a bug?

Comment (tarekgh) [reply release-notes/8.0/preview/api-diff/preview7/Microsoft.NETCore.App/8.0-preview7_System.Diagnostics.Metrics.md:11]: The move is expected. If we are documenting the API diff difference should be noted as moved and not introduced APIs.

File: release-notes/8.0/preview/api-diff/preview7/Microsoft.WindowsDesktop.App/8.0-preview7_Microsoft.Win32.md
Comment (dipeshmsft) [inline release-notes/8.0/preview/api-diff/preview7/Microsoft.WindowsDesktop.App/8.0-preview7_Microsoft.Win32.md:1]: LGTM for WPF diffs

**Review by dipeshmsft — APPROVED** [review]

**Comment by dreddy-work** [comment]
WinForms LGTM.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**